packages/did/src/DidDetails/DidDetails.ts

```diff
@@ -70,33 +65,25 @@ export function getEndpoint(
 /**
  * Generate a signature over the provided input payload, either as a byte array or as a HEX-encoded string.
  *
- * @param did The DID data.
+ * @param did The DID to be used.
  * @param payload The byte array or HEX-encoded payload to sign.
  * @param sign The sign callback to use for the signing operation.
- * @param keyId The key ID to use to generate the signature.
  *
  * @returns The resulting [[DidSignature]].
  */
 export async function signPayload(
-  did: Partial<DidDocument> & Pick<DidDocument, 'authentication' | 'uri'>,
+  did: DidUri,
   payload: Uint8Array | string,
-  sign: SignCallback,
-  keyId: DidVerificationKey['id']
+  sign: SignCallback
 ): Promise<DidSignature> {
-  const key = getKey(did, keyId)
-  if (!key || !verificationKeyTypes.includes(key.type)) {
-    throw new SDKErrors.DidError(
-      `Failed to find verification key with ID "${keyId}" on DID "${did.uri}"`
-    )
-  }
-  const alg = signatureAlgForKeyType[key.type as VerificationKeyType]
-  const { data: signature } = await sign({
-    publicKey: key.publicKey,
-    alg,
+  const { data: signature, keyUri } = await sign({
     data: Crypto.coToUInt8(payload),
+    keyRelationship: 'authentication',
```

Comment:
oh, I was under the impression, that we do, because of the previous type for the did:
```
Partial<DidDocument> & Pick<DidDocument, 'authentication' | 'uri'>
```
This was asking for a did document, which has an uri and authentication key.

I can add the keyRelationship as an optional parameter, defaulting to `authentication`, though, if this functions is supposed to be used with other key relationships, too.

Reply:
I think it makes sense. I can't say for sure though, because when it comes to non-extrinsic signing I just don't know which key we use for what